pkg/api/graph.go

```diff
@@ -107,7 +107,7 @@ func BuildGraph(steps []Step) []*StepNode {
 				for _, otherCreates := range other.Step.Creates() {
 					if nodeRequires.Matches(otherCreates) {
 						isRoot = false
-						other.Children = append(other.Children, node)
```

Comment:
How did you end up with nodes in `allNodes` more than once?

Reply:
A step link parameter that depended on release tags and a regular Requires() that depended on release tags.